src/transformers/testing_utils.py

```diff
@@ -1615,3 +1616,38 @@ def new_request(self, method, **kwargs):
             self.other_request_count += 1
 
         return self.old_request(method=method, **kwargs)
+
+
+# adapted from https://stackoverflow.com/questions/32163436/python-decorator-for-printing-every-line-executed-by-a-function
+class set_reproducible(ContextDecorator):
```

Comment:
This is interesting, does that mean we can use this decorator every time we want to test PyTorch models that involve stochasticity? Cause ViTMAE is another example (as it creates a random boolean mask inside).

I'm currently using `torch.manual_seed` as seen [here](https://github.com/huggingface/transformers/blob/0f257a87749e0a72bda260c6f319a45dae1e7c4d/tests/models/vit_mae/test_modeling_vit_mae.py#L235). Should we use this decorator instead?

Reply:
I think this change could be well applied to `ViTMAE` - they have exactly the same stochasticity situation.

Reply:
It does work for the mentioned test 🥳